src/include/timestamp.i

```diff
@@ -0,0 +1,207 @@
+__wt_time_windows_equal(WT_TIME_WINDOW *tw1, WT_TIME_WINDOW *tw2)
+{
+    return (tw1->start_ts == tw2->start_ts && tw1->start_txn == tw2->start_txn &&
+      tw1->durable_start_ts == tw2->durable_start_ts && tw1->stop_ts == tw2->stop_ts &&
+      tw1->stop_txn == tw2->stop_txn && tw1->durable_stop_ts == tw2->durable_stop_ts &&
+      tw1->prepare == tw2->prepare);
+}
+
+/*
+ * __wt_time_window_set_start --
+ *     Set the start values of a time window from those in an update structure.
+ */
+static inline void
+__wt_time_window_set_start(WT_TIME_WINDOW *tw, WT_UPDATE *upd)
+{
+    tw->start_ts = upd->start_ts;
+    tw->start_txn = upd->txnid;
+    tw->durable_start_ts = upd->durable_ts;
+}
+
+/*
+ * __wt_time_window_set_stop --
+ *     Set the start values of a time window from those in an update structure.
+ */
+static inline void
+__wt_time_window_set_stop(WT_TIME_WINDOW *tw, WT_UPDATE *upd)
+{
+    tw->stop_ts = upd->start_ts;
+    tw->stop_txn = upd->txnid;
+    tw->durable_stop_ts = upd->durable_ts;
+}
+
+/*
+ * __wt_time_aggregate_init --
+ *     Initialize the fields in an aggregated time window to their defaults.
+ */
+static inline void
+__wt_time_aggregate_init(WT_TIME_AGGREGATE *ta)
+{
+    ta->oldest_start_ts = WT_TS_NONE;
+    ta->oldest_start_txn = WT_TXN_NONE;
+    /*
+     * The aggregated durable timestamp values represent the maximum durable timestamp over set of
+     * timestamps. These aggregated max values are used for rollback to stable operation to find out
+     * whether the page has any timestamp updates more than stable timestamp.
+     */
+    ta->newest_start_durable_ts = WT_TS_NONE;
+    ta->newest_stop_ts = WT_TS_MAX;
+    ta->newest_stop_txn = WT_TXN_MAX;
+    ta->newest_stop_durable_ts = WT_TS_NONE;
+
+    ta->prepare = 0;
+}
+
+/*
+ * __wt_time_aggregate_init_max --
+ *     Initialize the fields in an aggregated time window to maximum values, since this structure is
+ *     generally populated by iterating over a set of timestamps and calculating max/min seen for
+ *     each value, it's useful to be able to start with a negatively initialized structure.
+ */
+static inline void
+__wt_time_aggregate_init_max(WT_TIME_AGGREGATE *ta)
+{
+    ta->oldest_start_ts = WT_TS_MAX;
+    ta->oldest_start_txn = WT_TXN_MAX;
+    /*
+     * The aggregated durable timestamp values represent the maximum durable timestamp over set of
+     * timestamps. These aggregated max values are used for rollback to stable operation to find out
+     * whether the page has any timestamp updates more than stable timestamp.
+     */
+    ta->newest_start_durable_ts = WT_TS_NONE;
+    ta->newest_stop_ts = WT_TS_NONE;
+    ta->newest_stop_txn = WT_TXN_NONE;
+    ta->newest_stop_durable_ts = WT_TS_NONE;
+
+    ta->prepare = 0;
+}
+
+/*
+ * __wt_time_aggregate_is_empty --
+ *     Return true if the time aggregate is equivalent to the default time aggregate.
+ */
+static inline bool
+__wt_time_aggregate_is_empty(WT_TIME_AGGREGATE *ta)
```

Comment:
So empty means it is the default value? If that is the case, why not call it is_default? is_empty is confusing to me.

Reply:
I thought about this a bit, I don't feel very strongly, but I think I prefer `empty`. We could have various different versions of default over time - so that could age out. For example we already have both `time_window_init` and `time_window_init_max`.

Reply:
The default window is from NONE to MAX. What about `__wt_time_aggregate_is_max_window`?

Your argument makes sense to me. Just another suggestion. I'm fine with empty for now as well.

Reply:
I don't feel strongly, so I'll leave it as empty for now (and it's easy to change in the future).